src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1270,6 +1271,11 @@ private static DkmClrCompilationResultFlags GetLocalResultFlags(LocalSymbol loca
                     var instance = new DisplayClassInstanceFromParameter(parameter);
                     displayClassInstances.Add(new DisplayClassInstanceAndFields(instance));
                 }
+                else
+                {
+                    // Collect parameter names that will hide display class variables from outer scopes
+                    userDefinedParameterNames.Add(parameter.Name);
```

Comment:
>userDefinedParameterNames.Add(parameter.Name); [](start = 20, length = 46)

I am not sure if this is going to handle a scenario with a nested local function when an enclosing local function shadows a local rather than when the nested local function does that.

> For evaluating a specific expression, we are constructing a special binder

If that is the case, I think that the binder should handle the shadowing and it is not clear why any manual work is needed for that.

Reply:
> If that is the case, I think that the binder should handle the shadowing and it is not clear why any manual work is needed for that.

We use a single binder for all the display class variables, and that binder is added on top of the binder for method parameters. I'm not sure why that design was adopted (I agree that doesn't seem the best), but I'm not planning to redesign for purpose of this bug.

Reply:
Added a test for nested local function that captures parameter from enclosing local function that shadows outer scope. That works okay

---
In reply to: [574721376](https://github.com/dotnet/roslyn/pull/51129#discussion_r574721376) [](ancestors = 574721376)